tests/suites/test_suite_ssl.data

```diff
@@ -440,19 +440,19 @@ depends_on:MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_AES_C:
 handshake_ciphersuite_select:"TLS-ECDHE-ECDSA-WITH-AES-256-CCM":MBEDTLS_PK_ECDSA:"":PSA_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:PSA_KEY_USAGE_DERIVE:MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE:0
 
 Handshake, select ECDH-RSA-WITH-AES-256-CBC-SHA384, non-opaque
-depends_on:MBEDTLS_HAS_ALG_SHA_384_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_AES_C:MBEDTLS_RSA_C:MBEDTLS_ECDSA_C:MBEDTLS_ECDH_C:MBEDTLS_ECP_DP_SECP256R1_ENABLED:MBEDTLS_ECP_DP_SECP384R1_ENABLED:MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
+depends_on:MBEDTLS_HAS_ALG_SHA_384_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_AES_C:MBEDTLS_RSA_C:MBEDTLS_PK_CAN_ECDSA_SOME:MBEDTLS_ECDH_C:MBEDTLS_ECP_DP_SECP256R1_ENABLED:MBEDTLS_ECP_DP_SECP384R1_ENABLED:MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
```

Comment:
Ah, that works for now but I'm afraid that's not strict enough in principle, which might affect future work (that is, when we work on builds with only public or only private operations). For the handshake to complete, one party need to make a signature and the other party needs to verify it, and since we act as both parties in these tests, we need both, so that's `MBEDTLS_PK_CAN_ECDSA_SIGN:MBEDTLS_PK_CAN_ECDSA_VERIFY`. (That is, sign _and verify, not _or_.)

For now the difference is not visible in our tests because we make no attempt to support builds with only public or only private operations, but there's demand for that (at least for builds with only public operations) so hopefully we'll get there at some point. And when we do, it will help if we did our best to use the right dependencies here :)

Reply:
Is it worth to create a companion macro like `MBEDTLS_PK_CAN_ECDSA_FULL` to signal that there are both signing *and*  verifying capabilities?

Reply:
No sure. I think just listing them both is OK in tests: it's not much longer and those dependency lines are not short in the first place anyway. OTOH in other places it might be more convenient to have a single macro, for example in `ssl-opt.sh` to save vertical space by having on `requires_config_enabled` line instead of two.

Reply:
Ok, so I will keep this change for the [following PR](https://github.com/Mbed-TLS/mbedtls/pull/7082) if it is necessary ;)